src/main/groovy/com/morpheusdata/scvmm/ScvmmProvisionProvider.groovy

```diff
@@ -894,9 +895,27 @@ class ScvmmProvisionProvider extends AbstractProvisionProvider implements Worklo
 		}
 	}
 
+	def getNonRootStorageVolumesFromConfigs(workload) {
+		def configs = workload.configs
+		if (configs instanceof String) {
+			configs = new JsonSlurper().parseText(configs)
+		}
+		def volumesList = configs?.volumes ?: []
+		def storageVolumeProps = StorageVolume.metaClass.properties*.name as Set
+		def storageVolumes = volumesList.findAll { !it.rootVolume }.collect { volMap ->
+			def filteredVolMap = volMap.findAll { k, v -> storageVolumeProps.contains(k) }
+			if (filteredVolMap.id == -1) {
+				new StorageVolume(filteredVolMap)
+			}
+		}.findAll {it != null }
+		return storageVolumes
+	}
+
     def additionalTemplateDisksConfig(Workload workload, scvmmOpts) {
         // Determine what additional disks need to be added after provisioning
         def additionalTemplateDisks = []
+		def nonRootAdditionalVolumes = getNonRootStorageVolumesFromConfigs(workload)
```

Comment:
For documentation purposes, can you split this into two conditions:
- If its a cloning scenario , call getUserAddedVolumes
- If its a new VM creation scenario, retain the existing code. Basically `getContainerDataDiskList`, which fetches all data disks except for root volume.

Please make this change , even though the exisiting `getNonRootStorageVolumesFromConfigs` works for both scenarios, because it makes for more readable code and better documentation.